src/main/java/de/tum/in/www1/artemis/repository/BuildJobRepository.java

```diff
@@ -0,0 +1,26 @@
+package de.tum.in.www1.artemis.repository;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import de.tum.in.www1.artemis.domain.BuildJob;
+
+public interface BuildJobRepository extends JpaRepository<BuildJob, Long> {
+
+    List<BuildJob> findAllByCourseId(Long courseId);
+
+    List<BuildJob> findAllByExerciseId(Long exerciseId);
+
+    List<BuildJob> findAllByParticipationId(Long participationId);
+
+    Optional<BuildJob> findFirstByParticipationIdOrderByBuildStartDateDesc(Long participationId);
+
+    List<BuildJob> findAllByBuildAgentAddress(String buildAgentAddress);
+
+    @Query("SELECT buildjob FROM BuildJob buildjob WHERE buildjob.dockerImage = :dockerImage ORDER BY buildjob.buildStartDate DESC")
```

Comment:
please use the syntax with multi line strings and make sure to use the correct indentation